server/datastore/mysql/migrations/tables/20220322091216_UpdateAppConfigLogEndpoint.go

```diff
@@ -0,0 +1,62 @@
+package tables
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20220322091216, Down_20220322091216)
+}
+
+func Up_20220322091216(tx *sql.Tx) error {
+	const selectStmt = `SELECT json_value FROM app_config_json LIMIT 1`
+	var raw json.RawMessage
+	var config fleet.AppConfig
+
+	row := tx.QueryRow(selectStmt)
+	if err := row.Scan(&raw); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil
+		}
+		return errors.Wrap(err, "select app_config_json")
+	}
+	if err := json.Unmarshal(raw, &config); err != nil {
+		return errors.Wrap(err, "unmarshal appconfig")
+	}
+
+	var (
+		oldPath = []byte(`"/api/v1/osquery/log"`)
+		newPath = []byte(`"/api/latest/osquery/log"`)
+		updated = false
+	)
+	if config.AgentOptions != nil {
+		oldOpts := []byte(*config.AgentOptions)
+		newOpts := json.RawMessage(bytes.ReplaceAll(oldOpts, oldPath, newPath))
+		config.AgentOptions = &newOpts
+		updated = !bytes.Equal(oldOpts, newOpts)
+	}
```

Comment:
@chiiph thoughts on that? Should the user migrate their appconfig explicitly at their own schedule, and we point this out in the upgrade / release docs, or should we have the migation to it automatically? I don't think anything would break by changing it?

Reply:
We should try to do as much as we can for the user. But this raises an issue we haven't taken into account: osquery will use the endpoints that are stored in the config first, and we have no way to update those... I think this will be worth discussing in the backend weekly.

Reply:
After some back and forth the decision is to not version the osquery endpoints. This means removing the `v1` portion of it and adding an alias with the `v1` so the change is backwards compatible for the moment. In the future we'll just remove that alias.

The reasoning is that the osquery endpoints won't change given how that has happened in the past (osquery hasn't changed the way it expects to talk to servers in the past 3 years or so).